SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -191,6 +190,29 @@ int ASpatialFunctionalTest::GetNumberOfClientWorkers()
 	return Counter;
 }
 
+void ASpatialFunctionalTest::GatherRelevantActors(TArray<AActor*>& OutActors)  const
+{
+	if (ObservationPoint)
+	{
+		OutActors.AddUnique(ObservationPoint);
+	}
+
+	TArray<AActor*> FoundActors;
+	UGameplayStatics::GetAllActorsOfClass(GetWorld(), AActor::StaticClass(), FoundActors);
+
+	for (int32 ActorIndex = 0; ActorIndex < FoundActors.Num(); ++ActorIndex)
+	{
+		AActor* FoundActor = FoundActors[ActorIndex];
+		UActorComponent* AutoDestroyComponent = FoundActor->FindComponentByClass<UAutoDestroyComponent>();
+		if (AutoDestroyComponent != NULL)
+		{
+			OutActors.AddUnique(FoundActor);
+		}
+	}
+
+	OutActors.Append(DebugGatherRelevantActors());
+}
+
```

Comment:
It would be good to know where this is used, I'm afraid that if it is called often that it will be very slow and maybe shouldn't be overriden. 

Also, you don't need UGameplayStatics btw, you can do:
for( TActorIterator<AActor> It(GetWorld()); It; ++It )

which prevents allocating an array to have all the actors. Those UGameplayStatics are more of convenience functions to help BPs do things

Reply:
It's called in the constructor of FFTestRenderingSceneProxy
![image](https://user-images.githubusercontent.com/64085832/87547078-3fcbfc00-c6b3-11ea-98d1-63e8e337f711.png)

And also in the function UFuncTestRenderingComponent::CalcBounds
![image](https://user-images.githubusercontent.com/64085832/87547165-5f632480-c6b3-11ea-8674-35978baf61b0.png)

Reply:
Using a print statement showed that this function was called 40 times in one test.

Reply:
Assuming the test is fast, this probably means ticking each frame. So I'd dis-advise against overriding the function to collect them then, it would be good to see what they do with it, specially the auto destroy actors array, if they use it or simply for some bound checks.

Reply:
I've now checked with a breakpoint and this function is only called on registering and destroying components, so I think it is ok.